### PR TITLE
feat(psim)!: preapre settings to launch localization modules on psim

### PR DESCRIPTION
**Pull request by yuki-takagi-66**
## Description
This PR is a part of the following issue:
https://github.com/autowarefoundation/autoware_launch/issues/1100

This PR allows localization and simple_planning_simulator to support launching the localization module in psim.
launch PR: https://github.com/autowarefoundation/autoware_launch/pull/1094

For more details, please see the launch PR and the issue page.

## How was this PR tested?
psim tests and scenario simulator

## Notes for reviewers

None.

### CODE REVIEW

**Review by takayuki5168 — APPROVED** [review]

File: launch/tier4_simulator_launch/launch/simulator.launch.xml [hunk 1]
```diff
@@ -129,16 +135,61 @@
...
+        <!-- start name space localization -->
+        <push-ros-namespace namespace="localization"/>
+        <group>
+          <push-ros-namespace namespace="util"/>
+          <include file="$(find-pkg-share pose_initializer)/launch/pose_initializer.launch.xml">
+            <arg name="user_defined_initial_pose/enable" value="false"/>
+            <arg name="user_defined_initial_pose/pose" value="[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]"/>
+            <arg name="ndt_enabled" value="false"/>
+            <arg name="gnss_enabled" value="false"/>
+            <arg name="ekf_enabled" value="true"/>
```
Comment (kosuke55) [inline launch/tier4_simulator_launch/launch/simulator.launch.xml:161]: >  localization modules without NDT will be launched.

not NDT but ekf?

Comment (kosuke55) [reply launch/tier4_simulator_launch/launch/simulator.launch.xml:161]: this is just pose_initializer's args. sorry I misunderstood

**Review by kosuke55 — APPROVED** [review]

File: launch/tier4_simulator_launch/launch/simulator.launch.xml [hunk 2]
```diff
@@ -129,16 +135,61 @@
...
+        <arg name="user_defined_initial_pose/enable" value="false"/>
+        <arg name="user_defined_initial_pose/pose" value="[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]"/>
+        <arg name="ndt_enabled" value="false"/>
+        <arg name="gnss_enabled" value="false"/>
+        <arg name="ekf_enabled" value="false"/>
+        <arg name="yabloc_enabled" value="false"/>
+        <arg name="stop_check_enabled" value="false"/>
+        <arg name="config_file" value="$(var pose_initializer_param_path)"/>
+      </include>
+    </group>
```
Comment (TakaHoribe) [inline launch/tier4_simulator_launch/launch/simulator.launch.xml:149]: Why is this code necessary?

**Review by yukkysaito — CHANGES_REQUESTED** [review]
Same comment
https://github.com/autowarefoundation/autoware_launch/pull/1094#pullrequestreview-2203877252

**Comment by yuki-takagi-66** [comment]
@yukkysaito 
Thank you.
As we discussed internally, we need to achieve the following two goals: 
"maintain a simulator environment whose only dependent module is planning/control" and "evaluate planning/control modules, considering the characteristics of localization in a closed loop."
To achieve these two goals, the preceding PR https://github.com/autowarefoundation/autoware_launch/pull/1106 provided a mode-switching feature supporting three or more states.

**Review by TakaHoribe — APPROVED** [review]

**Review by yukkysaito — APPROVED** [review]

**Pull request merged**